event-app-ui/src/components/Profile/Profile.jsx

```diff
@@ -111,23 +127,92 @@ export default function Profile() {
             {isOpen ? (
               <Popup setIsOpen={setIsOpen} username={username} />
             ) : (
-              <div className="event-view">
-                {eventData.map((value, key) => {
-                  if (value.username === username) {
-                    return (
-                      <div key={key} className="event">
-                        <h2> {value.eventname}</h2>
-                        <p> {value.description}</p>
-                        <p>{value.location}</p>
-                        <p> {value.starttime}</p>
-                        <p> {value.endtime}</p>
-                        <button>More Details</button>
-                      </div>
-                    );
-                  }
-                  return null;
-                })}
-              </div>
+              <>
+                <h2 className="future-past">Future Events</h2>
+                <div className="event-view">
+                  {futureEvent.map((value, key) => {
+                    if (value.username === username) {
```

Comment:
From the previous diff, it's not ideal to filter events on the front end. I expect this logic to be fixed in a subsequent PR. However, in general L131-171 and L172-211 are pretty much the same so you can generate both using a .map() function too. No action needed for now, just FYI